packages/test/test-end-to-end-tests/src/test/gc/gcDatastoreAliased.spec.ts

```diff
@@ -4,99 +4,50 @@
 
 /**
  * Validates this scenario: When a datastore is aliased that it is considered a root datastore and always referenced
  */
 describeNoCompat("GC Data Store Aliased", (getTestObjectProvider) => {
     let provider: ITestObjectProvider;
-    const dataObjectFactory = new DataObjectFactory(
-        "TestDataObject",
-        TestDataObject,
-        [],
-        []);
-
-    const runtimeOptions: IContainerRuntimeOptions = {
-        summaryOptions: {
-            summaryConfigOverrides: {
-                state: "disabled",
-            },
-        },
-        gcOptions: {
-            gcAllowed: true,
-        },
-    };
-    const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
-        runtime.IFluidHandleContext.resolveHandle(request);
-    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
-        dataObjectFactory,
-        [
-            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
-        ],
-        undefined,
-        [innerRequestHandler],
-        runtimeOptions,
-    );
-
-    // Enable config provider setting to write GC data at the root.
-    const settings = { "Fluid.GarbageCollection.LogUnknownOutboundReferences": "true" };
-    const configProvider = mockConfigProvider(settings);
-
     let container1: IContainer;
     let container2: IContainer;
-    let mainDataStore1: TestDataObject;
-    let mainDataStore2: TestDataObject;
+    let mainDataStore1: ITestDataObject;
+    let mainDataStore2: ITestDataObject;
 
-     async function summarizeOnContainer(container: IContainer) {
-        const dataStore = await requestFluidObject<TestDataObject>(container, "default");
-        return dataStore.containerRuntime.summarize({ runGC: true, trackState: false });
+    async function waitForSummary(container: IContainer) {
+        const dataStore = await requestFluidObject<ITestDataObject>(container, "default");
+        return (dataStore._context.containerRuntime as ContainerRuntime).summarize({ runGC: true, trackState: false });
     }
 
-    const createContainer = async (): Promise<IContainer> => {
-        return provider.createContainer(runtimeFactory, { configProvider });
-    };
-    const loadContainer = async (): Promise<IContainer> => {
-        return provider.loadContainer(runtimeFactory, { configProvider });
-    };
     beforeEach(async () => {
-        provider = getTestObjectProvider();
-
-        // Create a Container for the first client.
-        container1 = await createContainer();
-        container2 = await loadContainer();
-
-        mainDataStore1 = await requestFluidObject<TestDataObject>(container1, "default");
-        mainDataStore2 = await requestFluidObject<TestDataObject>(container2, "default");
-        await provider.ensureSynchronized();
+        provider = getTestObjectProvider({ syncSummarizer: true });
+        container1 = await provider.makeTestContainer(defaultGCConfig);
```

Comment:
What's the difference between `provider.createContainer` and `provider.makeTestContainer`? Simplicity of reading?

Reply:
`createContainer` accepts a container runtime factory as input whereas `makeTestContainer` accepts options and creates container runtime factory. The disadvantage of using `createContainer` is that you don't get older versions of container runtime / data store runtime when running back-compat tests.